docs/explanation/parts.rst

```diff
@@ -0,0 +1,263 @@
+define behaviour that is specific to the type of project that the plugin
+supports. For example, the :py:mod:`cmake plugin <craft_parts.plugins.cmake_plugin>` provides the ``cmake-parameters`` and
+``cmake-generator`` keys that can be used to configure how :command:`cmake`
+is used in the build process.
+
+The build process can be further customised with the ``build-environment``,
+``build-attributes`` and ``override-build`` keys.
+
+The ``build-environment`` key defines assignments to shell environment variables in the build environment, specified as a list of key-value pairs.
+In YAML format, this can take either of the following forms:
+
+.. code:: yaml
+
+   build-environment: [MESSAGE: "Hello world", NAME: "Craft Parts"]
+
+   build-environment:
+     - MESSAGE: "Hello world"
+     - NAME: "Craft Parts"
+
+The ``build-attributes`` key allows a number of standard customisations to be
+applied to the build. Some of these are used to address issues that occur in
+specific situations; others, such as ``debug`` are generally useful.
+
+By default, Craft Parts builds independent parts in parallel. This can be
+disabled by setting the ``disable-parallel`` key to ``True``.
+
+The ``override-build`` key is used to override the build process provided by
+a plugin, and it can be used to replace or extend it.
+
+Build products
+~~~~~~~~~~~~~~
+
+After a part has been built in the *build* step, the build products are
+prepared in three further steps.
+
+If the ``organize`` key is specified, it is used to move files
+specify a list of mappings between files
+in the part key-value pairs
+
+   * How to manage the different steps in the build process
+
+    override-stage string
+    override-prime string
+
+
+
+   * Filesets
+
+ * How artifacts are exported/installed
+
+
+
+
+    stage-snaps ref: #/definitions/grammar-array
+    stage-packages ref: #/definitions/grammar-array
+    organize
+    filesets
+    stage array with at least 1 unique item(s) of string
+    prime array with at least 1 unique item(s) of string
+
+..    parse-info array with at least 1 unique item(s) of string
+
+.. _build-order:
+
+Defining the build order
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, when more than one part is specified, they are built in alphabetical order unless there are dependencies between parts.
+
+One way to define a dependency for a part is to use the ``after`` key in
+a part's definition to specify a list of parts that it will be built after. The parts whose names are supplied in the list will be *built and staged*
+before the part is built.
+
+This is covered in detail in :ref:`part_processing_order`.
+
+.. Overriding aspects of the build
+.. ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+How parts are built
+-------------------
+
+As described in :ref:`lifecycle`, parts are built in a sequence of steps:
+*pull*, *overlay*, *build*, *organize*, *stage* and *prime*.
+
+A part is built in a clean environment to ensure that only the base and its
```

Comment:
This explanation is a bit vague. Might again be worth breaking it into steps, e.g.: container launched with X os, build environment prepped, home user directory presented inside the container, etc.